cpp/src/io/utilities/column_buffer.cpp

```diff
@@ -101,9 +101,10 @@ void copy_buffer_data(string_policy const& buff, string_policy& new_buff)
 }  // namespace
 
 template <class string_policy>
-void column_buffer_base<string_policy>::create(size_type _size,
-                                               rmm::cuda_stream_view stream,
-                                               rmm::device_async_resource_ref mr)
+void column_buffer_base<string_policy>::create_with_mask(size_type _size,
+                                                         cudf::mask_state null_mask_state,
```

Comment:
there's an overlap between null_mask_state and is_nullable - e.g. there are invalid combinations. I wonder if null_mask_state should replace is_nullable, as it's a superset of states. I'm just not sure if null_mask_state is available when we call the ctor.

Reply:
Discussed offline, there aren't invalid combinations, the meaning of null_mask_state is actually "if there's a null mask, initialize it to this", so it's all good.